activerecord/lib/active_record/associations/has_many_through_association.rb

```diff
@@ -212,6 +212,38 @@ def delete_through_records(records)
           end
         end
 
+        def delete_through_record(record, through_records, method)
+          case method
+          when :destroy_async
+            has_other_owners_on_through_association = false
```

Comment:
I need a little help with this. I'm not sure what is the best way of recognizing that a given "has many through" record is not shared with any other "owner", and thus, can be deleted.

/cc @ghiculescu

Reply:
I'm not very familiar with this code, I would have thought that you'd be acting on the same records if you use `:destroy_async` or `:destroy` - the key difference being if it runs inline or in a background job.

Reply:
👍. The PR is ready for review.